pkg/chunked/filesystem_linux.go

```diff
@@ -304,7 +304,7 @@ func openFileUnderRoot(dirfd int, name string, flags uint64, mode os.FileMode) (
 	if errors.Is(err, unix.ENOENT) && hasCreate {
 		parent := filepath.Dir(name)
 		if parent != "" {
-			newDirfd, err2 := openOrCreateDirUnderRoot(parent, dirfd, 0)
+			newDirfd, err2 := openOrCreateDirUnderRoot(dirfd, parent, 0)
```

Comment:
On a related topic, wouldn't it make sense to create a struct wrapper for the dirfd, and have things be methods on it?

In my Rust code, we *extensively* use https://docs.rs/cap-std/latest/cap_std/ which is related, and has a `Dir` struct with a lot of methods, but in implementation it's just:

https://github.com/bytecodealliance/cap-std/blob/1dc17d327bdcd44a0f5ec53d9fc671285be55b34/cap-std/src/fs/dir.rs#L48-L50

Reply:
we need an explicit `mode` to support `ForceMask`